### ISSUE TITLE
patch_sdk tool must return non-zero error code if it reported any nits/warnings/errors

### ISSUE THREAD

**Issue by zanderso** [dart-lang/sdk#30127]
E.g.:

```
[4737/17343] ACTION //flutter/lib/snapshot:flutter_patched_sdk(//build/toolchain/fuchsia:x64)
Not enabling colors, stdout does not support ANSI colors.
flutter_patched_sdk/lib/ui/ui.dart: Nit: 'Symbol' is imported from both 'dart:_internal' and 'dart:core'.
```

See GN guidelines here: https://github.com/dart-lang/sdk/wiki/Building-with-GN#hints-suggestions-etc

In particular: "A successful GN build produces no output."

/cc @sigmundch

**Comment by sigmundch** [comment]
Seems like we added a new `nit` in fasta that I believe shouldn't be reported. I can see this also when building patched_dart2js_sdk in the regular Dart repo.

@peter-ahe-google  - is this Nit actually appropriate in this case? I believe we shouldn't report it when one of the libraries is `dart:core` (since the spec had a special rule that allows an import to shadow anything from `dart:core`)

**Comment by peter-ahe-google** [comment]
The "nit" is correct, it tells you about a potential source of confusion.

However, nits are turned off by default, and I've been wondering why they're showing up. They don't show up when we build platform.dill for the VM.

**Comment by lrhn** [comment]
For the record: The original spec allows a non-system library to implicitly shadow a member of a system-library by hiding the name in the system library import. It's not just `dart:core`, but all system libraries.
The rule only applies if the name is in the top-level scope and it is actually used, and then it also causes a warning.
That rule also forgot to account for the case where the name refers to the same original declaration, in which case there should be no conflict.

In this case, both libraries are `dart:`-libraries, so if we read that rule literally, and the name was actually used, the name would be hidden in both imports and you'd get two warnings.

We are considering removing the warning from that rule (and tightening up the wording) for Dart 2/strong mode.

**Comment by zanderso** [comment]
Unless they are indicating a fatal error, tools called from the build should swallow all command line messages. We'll need to roll Dart into Flutter and Fuchsia at least once this week. Could this issue please be addressed soon? Thanks!

**Comment by peter-ahe-google** [comment]
Sorry, I thought this was about the warnings from dart2js.

flutter_patched_sdk/lib/ui/ui.dart needs to be changed like this:

import 'dart:_internal' hide Symbol;

**Comment by zanderso** [comment]
I am not on the front-end team, so reassigning to @sigmundch. Re-adding the front-end label, because this is a bug in the front-end.

**Comment by peter-ahe-google** [comment]
This is not a bug, and I've told you how to fix the problem.

**Comment by zanderso** [comment]
Please do not close this issue until the problem has been addressed. I would personally rather revert the changes that led to this problem than editing code that I don't understand and have never touched before. Please feel free to re-assign to me if that is the resolution you would like.

**Comment by mraleph** [comment]
I am gonna fix it.

**Comment by mraleph** [comment]
I opened PR against engine. I don't know who has approval rights. @zanderso can you help me to find an appropriate reviewer?

**Comment by mraleph** [comment]
We have fixed this particular problem but we should also update patch_sdk tool to treat any warnings/nits/errors it reports appropriately and exit with non-zero error code - indicating that compilation of platform.dill failed. 

@zanderso indicated that guidelines for builds state that successful builds *must* produce no output. Conversely if patch_sdk produced any output (e.g. reported any nits) it must also treat this as a failure and return an appropriate exit code. 

@sigmundch would you have cycles to look at this?

**Comment by peter-ahe-google** [comment]
Proposed fix: [CL 2982443002](https://codereview.chromium.org/2982443002/).

When I discussed this with @zanderso, he pointed out that this was a recent regression in the front-end. After knowing that it was easy for @sigmundch and I to pin point the root cause.

However, I still think it makes sense to change the import of dart:_internal in ui.dart.

**Comment by peter-ahe-google** [comment]
Sorry, I'm rereading the original report.

I'll make the change to the exit code as well.
